src/app/public/modules/wait/wait-adapter.service.ts

```diff
@@ -21,13 +36,69 @@ export class SkyWaitAdapterService {
     waitEl: ElementRef,
     isFullPage: boolean,
     isWaiting: boolean,
+    id: string,
     isNonBlocking = false
-  ): void {
+  ) {
     let busyEl = isFullPage ? document.body : waitEl.nativeElement.parentElement;
     let state = isWaiting ? 'true' : undefined;
 
     if (!isNonBlocking) {
       this.renderer.setElementAttribute(busyEl, 'aria-busy', state);
+
+      if (isWaiting) {
+        // Propagate tab navigation if attempted into waited element
+        if (!isFullPage) {
+          let focusListenerFunc = this.renderer.listen(this.windowRef.getWindow(), 'keyup', (event: any) => {
```

Comment:
Why are we doing `keyup` on element waits, but doing `keydown` on full-page waits?

Reply:
Keydown can trigger multiple events while the key is held down. Smallest amount I can trigger manually is 3 just clicking the tab key. So keydown is safe to use in the other situation, because we are preventing the events for full page and not performing any actions. Using key up ensures that the event only fires once and we don't have the focus shift triggered multiple times for the same keypress.